### PR TITLE
Adds a config pattern

### PR DESCRIPTION
**Pull request by tjduigna**
Something to get the ball rolling.

### CODE REVIEW

File: exa/__init__.py [hunk 1]
```diff
@@ -6,18 +6,73 @@
...
+    @property
+    def log(self):
+        name = '.'.join([
+            self.__module__, self.__class__.__name__
+        ])
+        return logging.getLogger(name)
+
+    def __init__(self, *args, **kws):
+        kws.pop('config', None)
+        super().__init__(
```
Comment (avmarchenko) [inline exa/__init__.py]: This implies that exa 1.0 will only support python 3. I'm good with this change.

File: exa/__init__.py [hunk 2]
```diff
@@ -6,18 +6,73 @@
...
 """
 import os
-import tempfile
+import sys
 import logging.config
+
 import yaml
+from traitlets import Unicode, default, validate
+from traitlets.config import Configurable, Application
+
```
Comment (avmarchenko) [inline exa/__init__.py]: I think we need to import `Container` and some other imports (or at least disable tests for now).

File: exa/__init__.py (hunk 1 above)
Comment (tjduigna) [reply exa/__init__.py]: Right, that was initially just an oversight but I think that in 2015 it was important to keep py2 compat but in 2020 less so.

File: exa/__init__.py (hunk 2 above)
Comment (tjduigna) [reply exa/__init__.py]: Yeah I will add the CI pipeline as an issue and disable it for now.

**Pull request merged**